lib/handlebars/compiler/javascript-compiler.js

```diff
@@ -11,29 +11,45 @@ JavaScriptCompiler.prototype = {
   // PUBLIC API: You can override these methods in a subclass to provide
   // alternative compiled forms for name lookup and buffering semantics
   nameLookup: function(parent, name /* , type*/) {
+    const actual = _actualLookup();
+    const dangerousProperties = ['__defineGetter__','__defineSetter__','__lookupGetter__','__proto__'];
+
     // Do not allow to access constructor of any object/class
+    // See: https://snyk.io/vuln/SNYK-JS-HANDLEBARS-469063
     if (name === 'constructor') {
-      return parent + '.propertyIsEnumerable(\'constructor\') ? ' + parent + '.constructor : undefined';
-    }
-    var wrap,
-        ret;
-    if (parent.indexOf('depth') === 0) {
-      wrap = true;
+      return 'Object.prototype.hasOwnProperty.call(' + parent + ',\'constructor\') ? ' + actual + ' : undefined';
     }
 
-    if (/^[0-9]+$/.test(name)) {
-      ret = parent + "[" + name + "]";
-    } else if (JavaScriptCompiler.isValidJavaScriptVariableName(name)) {
-      ret = parent + "." + name;
-    }
-    else {
-      ret = parent + "['" + name + "']";
+    // Block the above dangerous properties altogether from being used. If they are used in a template, we assume it
+    // could be to exploit the lib since the keywords don't make sense for actual template compilation or rendering
+    // unlike 'constructor' which could be someone's occupation :) lol
+    // See https://snyk.io/vuln/SNYK-JS-HANDLEBARS-534988
+    if (dangerousProperties.includes(name)) {
```

Comment:
`includes` won't work in IE, is this our way of officially saying goodbye to IE support? 😄

Reply:
haha, as per Bill, we stopped support for IE as of 10.3. 😆